src/App.tsx

```diff
@@ -1,56 +1,31 @@
-import { BrowserRouter, Route, Routes, Link } from 'react-router-dom';
-import { AppBar, Toolbar, Button, Select, MenuItem } from '@mui/material';
-import { IntlProvider } from 'react-intl';
-import { FormattedMessage } from 'react-intl';
-import { useLanguage } from './components/LanguageContext';
-import { messages, Language } from './assets/translations/messages';
-import ListPage from './pages/ListPage';
-import ItemForm from './components/ItemForm';
-import EditItemPage from './pages/EditItemPage';
-import ListData from './pages/ListData';
-import { RecoilRoot } from 'recoil';
+import { BrowserRouter, Route, Routes } from "react-router-dom";
+import ListPage from "./pages/ListPage";
+import AddItemPage from "./pages/AddItemPage";
+import EditItemPage from "./pages/EditItemPage";
+import ListData from "./pages/ListData";
+import { RecoilRoot } from "recoil";
+import Layout from "./components/Layout";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
 
 const App = () => {
-  const { language, setLanguage } = useLanguage();
+  const queryClient = new QueryClient();
```

Comment:
This will re-instantiate every time the component re-renders, such as when changing the language. But you only need one instance for the whole lifetime of the app, so you can move this outside the component and React tree entirely and keep it as a constant. This way the cache etc will not reset.

You could see this in action if you render <ReactQueryDevtools initialIsOpen={false} /> within the `QueryClientProvider`, which shows a clickable logo at the bottom of the screen where you can track all the queries made. So you'll see the 'items' get cached, and then if you switch the language, you'll see it get reset because of the new `queryClient` instance.